config/rbac/role.yaml

```diff
@@ -20,8 +20,10 @@ rules:
   resources:
   - secrets
   verbs:
+  - delete
```

Comment:
why is this needed?

Reply:
@deepakm-ntnx The secret is created by the cluster-template. When the nutanixcluster object is created, it will add an owner ref to the secret and also remove the secret if nutanixcluster object is deleted. In order to delete the owned secret this permission is required.